### ISSUE TITLE
"Microsoft Edge could not be found."

### ISSUE THREAD

**Issue by RokeJulianLockhart** [microsoft/vscode-edge-devtools#1984]
1.	## **Environment (please complete the following information)**
	
	**                                   | **
	---------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------
	**OS**            | cpe:/o:opensuse:tumbleweed:20240125
	**Extension version**  | https://marketplace.visualstudio.com/_apis/public/gallery/publishers/ms-edgedevtools/vsextensions/vscode-edge-devtools/2.1.4/vspackage

	

1.	## **Describe the bug**
	
	> Microsoft Edge could not be found. Ensure you have installed Microsoft Edge and that you have selected 'default' or the appropriate version of Microsoft Edge in the extension settings panel.

	...yet:

	> ![20240131T185423GMT](https://github.com/microsoft/vscode-edge-devtools/assets/42837531/51e21a8b-0f92-4564-b576-ad6e359ca3b1)
	
1.	## **Reproduction**
	
	1.	Install cpe:/o:opensuse:tumbleweed:20240125. (https://forums.opensuse.org/t/permalink-to-specific-iso/172140)
	1.	```.PS1
		flatpak install 'https://dl.flathub.org/repo/appstream/com.microsoft.EdgeDev.flatpakref' # https://github.com/flatpak/flatpak/issues/3097#issuecomment-1484796387
		flatpak update --commit=ce6eb46d90027c94dd3aa77fccab81a9d8cd849695c0ebd6e29bbda239b23144 com.microsoft.EdgeDev # https://unix.stackexchange.com/a/552700/386242
		```
	1.	```.PS1
		Invoke-WebRequest -URI 'https://marketplace.visualstudio.com/_apis/public/gallery/publishers/ms-edgedevtools/vsextensions/vscode-edge-devtools/2.1.4/vspackage' -OutFile 'ms-edgedevtools.vscode-edge-devtools-2.1.4.vsix' # https://stackoverflow.com/questions/57912072/invoke-webrequest-without-outfile#comment137383603_57912181
		code-insiders --install-extension 'ms-edgedevtools.vscode-edge-devtools-2.1.4.vsix' # https://stackoverflow.com/a/34339780/9731176
		```
		The configuration of the extension is its default. (https://github.com/microsoft/vscode/issues/203923#issue-2110803487)
	1.	Enable `ms-edgedevtools.vscode-edge-devtools`. 
	1.	Launch.
	
1.	## **Expected behaviour**
	
	Microsoft Edge should appear.
	
1.	## **Additional context**
	
	It used to, a few weeks ago.

**Comment by captainbrosset** [comment]
Thank you for the detailed bug report. I'll add this to our backlog and let our team know so they can investigate.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/microsoft/vscode-edge-devtools/issues/1984#issuecomment-1923319012

Turns out that it works with https://packages.microsoft.com/yumrepos/edge/microsoft-edge-stable-121.0.2277.98-1.x86_64.rpm?brand=M102, so I've reported this at https://github.com/flathub/com.microsoft.Edge/issues/448#issue-2115261259 instead.

**Comment by captainbrosset** [comment]
Let me tag @ElormCoch who, I know, has already been looking into this.

**Comment by RokeJulianLockhart** [comment]
Per https://github.com/flathub/com.microsoft.Edge/issues/448#issuecomment-1952941976, I expect that this must be remediated here or https://stackoverflow.com/a/78022614/9731176.

**Comment by ElormCoch** [comment]
Thanks for the feedback.

I did some digging into why the issue happens and it's because the devtools extension expects Edge to be installed in this path on linux distributions `/opt/microsoft/msedge-[dev/canary/stable]/msedge`

The installation via flatpak shows the executable path as `/app/extra/msedge` when you navigate to `edge://version` in the browser. But in fact, this seems to be a pseudo path, since this path doesn't exist on disk, and the actual location of the browser binary is in this path `/var/lib/flatpak/app/com.microsoft.EdgeDev/x86_64/stable/..../files/extra/msedge`


On the devtools extension side, we've considered providing a way to have folks specify a browser path in a setting: 
However, this new setting will conflict with the current "channel" setting  (dev, stable, channel) and also open up more errors such as users specifying a non-edge browser, a malicious binary..


A couple workarounds I tried:
* Changing the flatpak install location following instructions here - [](https://docs.flatpak.org/en/latest/tips-and-tricks.html#adding-a-custom-installation). This unfortunately installs the browser in a path like `opt/microsoft/msedge-dev/app/com.microsoft.EdgeDevx86_64/dev/.../files/extra/msedge` which isn't what is expected but maybe there's another way to do this i'm unaware of.

* Creating a symlink between the actual browser install location and the expected path - e.g. `ln -s /actual/path/to/edge /opt/microsoft/msedge/msedge` (for stable). This unblocks the scenario where you're unable to launch an instance of the devtools extension and the edge browser. You however would need to type in the url to your development project in the browser manually.

I hope this helps.
